src/backend/distributed/commands/table.c

```diff
@@ -4154,6 +4154,15 @@ ConvertNewTableIfNecessary(Node *createStmt)
 				return;
 			}
 
+			/*
+			 * We allow mat views in a distributed schema but do not make them a tenant
+			 * table. We should skip converting them.
+			 */
```

Comment:
again, why not (!IsRegularTable()) check here?

Reply:
We should fail instead of silently allowing those tables under distributed schema. `CreateTenantSchema` does required checks and fails.